lms/resources/_js_config/__init__.py

```diff
@@ -518,55 +519,62 @@ def _hypothesis_client(self):
                     "authority": self._authority,
                     "enableShareLinks": False,
                     "grantToken": grant_token,
-                    "groups": self._groups(),
                 }
             ]
         }
 
-    def _groups(self):
-        if self._context.grouping_type == Grouping.Type.COURSE:
-            return [self._context.course.groupid(self._authority)]
+    def _groups(self, grouping_type, course):
+        if grouping_type == Grouping.Type.COURSE:
+            return [course.groupid(self._authority)]
 
         # If not using the default COURSE grouping point the FE
         # to the sync API to dynamically get the relevant groupings.
         return "$rpc:requestGroups"
 
-    def _sync_api(self, assignment):
-        """Add configuration the front-end will use to get groupings."""
+    def _configure_groups(self, course, assignment):
+        """Configure how the client will fetch groups when in LAUNCH mode."""
+        if not self._context.application_instance.provisioning:
+            return
 
-        if self._context.grouping_type == Grouping.Type.COURSE:
-            return None
+        grouping_type = self._request.find_service(
+            name="grouping"
+        ).get_launch_grouping_type(self._request, course, assignment)
 
-        req = self._request
+        self._config["hypothesisClient"]["services"][0]["groups"] = self._groups(
+            grouping_type, course
+        )
 
-        return {
-            "authUrl": req.route_url(req.product.route.oauth2_authorize),
-            "path": req.route_path("api.sync"),
-            # This data is consumed by the view in `lms.views.api.sync` which
-            # defines the arguments it expects. We need to match that
-            # description. Anything we add here should be echoed back by the
-            # frontend.
-            "data": {
-                "resource_link_id": assignment.resource_link_id,
-                "lms": {
-                    "product": self._request.product.family,
-                },
-                "context_id": self._request.lti_params["context_id"],
-                "group_set_id": self._request.product.plugin.grouping.get_group_set_id(
-                    self._request, assignment
-                ),
-                "group_info": {
-                    key: value
-                    for key, value in self._request.lti_params.items()
-                    if key in GroupInfo.columns()
+        self._config["api"]["sync"] = None
+        if grouping_type != Grouping.Type.COURSE:
```

Comment:
The sync API and this `$rpc:requestGroups` thing are the same thing. Setting them up here together.